samza-core/src/main/java/org/apache/samza/runtime/AbstractApplicationRunner.java

```diff
@@ -106,4 +112,25 @@ final ExecutionPlan getExecutionPlan(StreamApplication app) throws Exception {
   final StreamManager getStreamManager() {
     return streamManager;
   }
+
+  /**
+   * Write the execution plan JSON to a file
+   * @param planJson JSON representation of the plan
+   */
+  final void writePlanJsonFile(String planJson) {
+    try {
+      String content = "plan='" + planJson + "'";
+      String binPath = System.getenv(ShellCommandConfig.JOB_BIN_DIR());
```

Comment:
hmm, you're right. log is a better place to put it. I was thinking maybe put them along with plan.html is easier since they can be copied over.  But I think I can let the html to cover both cases. Let me update the patch with this change.